src/operations/monochrome.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{error::MagickError, image::Image, wm_err};
+use image::{DynamicImage, GrayImage, Luma};
+
+pub fn monochrome(image: &mut Image) -> Result<(), MagickError> {
+    let mut grayscale = image.pixels.to_luma8();
+    apply_contrast(&mut grayscale, CONTRAST_FACTOR);
+    let noise_texture = NoiseTexture::load()?;
+    apply_dithering(&mut grayscale, &noise_texture);
+    image.pixels = DynamicImage::ImageLuma8(grayscale);
+
+    Ok(())
+}
+
+/// This is empirically tuned to give results similar to ImageMagick's -monochrome
+const CONTRAST_FACTOR: f32 = 10.0;
+
+fn apply_contrast(image: &mut GrayImage, contrast_factor: f32) {
```

Comment:
Ah, just saw https://docs.rs/image/latest/image/enum.DynamicImage.html#method.adjust_contrast, can try to replace this code with that.